supervised/automl.py

```diff
@@ -1087,11 +746,472 @@ def predict(self, X):
         elif self._ml_task == MULTICLASS_CLASSIFICATION:
             target_is_numeric = self._data_info.get("target_is_numeric", False)
             if target_is_numeric:
-                predictions["label"] = predictions["label"].astype(int)
+                predictions["label"] = predictions["label"].astype(np.int32)
             return predictions
+        # Regression
         else:
             return predictions
 
+    def _predict(self, X):
+
+        predictions = self._base_predict(X)
+        # Return predictions
+        # If classification task the result is in column 'label'
+        # If regression task the result is in column 'prediction'
+        return (
+            predictions["label"].to_numpy()
+            if self._ml_task != REGRESSION
+            else predictions["prediction"].to_numpy()
+        )
+
+    def _predict_proba(self, X):
+        # Check is task type is correct
+        if self._ml_task == REGRESSION:
+            raise AutoMLException(
+                f"Method `predict_proba()` can only be used when in classification tasks. Current task: '{self._ml_task}'."
+            )
+
+        # Make and return predictions
+        # If classification task the result is in column 'label'
+        # Need to drop `label` column.
+        return self._base_predict(X).drop(["label"], axis=1).to_numpy()
```

Comment:
In case of large ensembles, if user will need labels and probabilities, then he will need to run both `predict` and `predict_proba`. I think there should be `predict_all` method to predict probabilities and labels at once.

Reply:
In which format `predict_all` would return the data? The original Pandas dataframe?

Reply:
yes, pandas data frame

Reply:
The pandas dataframe returned by `_base_predict()` contains `prediction_0`, ..., `prediction_n`, `label`. Is this fine?
The method would be:
```
def _predict_all(self, X):
        # Check is task type is correct
        if self._ml_task == REGRESSION:
            raise AutoMLException(
                f"Method `predict_all()` can only be used when in classification tasks. Current task: '{self._ml_task}'."
            )

        # Make and return predictions
        return self._base_predict(X)
```